### ISSUE TITLE
Couldn't find tmdb id for media

### ISSUE THREAD

**Issue by daerup** [jorenn92/Maintainerr#1469]
**Describe the bug**
TMDb cannot fetch id for Die Hart. Die Hart is not the only movie this happens. Sometimes it works, sometime it doesn't. Is there maybe TMDb Rate Limiting at play? 

On another Note which I think might be related. The Id couldn't be populated and got resolved to 'undefined'. I couldn't reproduce this tho:
```
2024-12-26T19:56:10.327687543Z [maintainerr] | 26/12/2024 20:56:10  [DEBUG] [TmdbApiService] GET request to https://api.themoviedb.org/3/movie/undefined?api_key=[REDACTED]&language=en&append_to_response=credits,external_ids,videos,release_dates,watch%2Fproviders failed: AxiosError: Request failed with status code 404
```

**To Reproduce**
1. Add Die Hart to Radarr
2. Go into Collection and Test Media on Die Hart.

**Expected behavior**
Request could be made an proceed with other rules.

**Screenshots**
```
2024-12-26T20:13:47.474176209Z [maintainerr] | 26/12/2024 21:13:47  [INFO] [RulesService] Successfully updated rulegroup '[Movies] Requested & Ready'.
2024-12-26T20:14:45.572031605Z [maintainerr] | 26/12/2024 21:14:45  [DEBUG] [OverseerrGetterService] Couldn't find tmdb id for media 'Die Hart' with id '79387'. As a result, no Overseerr query could be made.
2024-12-26T20:14:45.572067652Z [maintainerr] | 26/12/2024 21:14:45  [DEBUG] [OverseerrGetterService] Couldn't fetch Overseerr metadate for media 'Die Hart' with id '79387'. As a result, no Overseerr query could be made.
```

**Additional context**
Exported rules:
```
mediaType: MOVIES
rules:
  - "0":
      - firstValue: Tautulli.lastViewedAt
        action: BEFORE
        customValue:
          type: custom_days
          value: "555"
  - "1":
      - operator: OR
        firstValue: Tautulli.addDate
        action: BEFORE
        customValue:
          type: custom_days
          value: "180"
      - operator: AND
        firstValue: Tautulli.viewCount
        action: EQUALS
        customValue:
          type: number
          value: 0
  - "2":
      - operator: AND
        firstValue: Overseerr.isRequested
        action: EQUALS
        customValue:
          type: boolean
          value: "true"
```

**Comment by ydkmlt84** [comment]
https://docs.maintainerr.info/Common/#tvdbid-or-tmdbid-errors

Maintainerr pulls the TVDB/TMDb ID from the items Plex metadata (which isn't always great). Most likely Plex has added the wrong TMDB id somehow. You could verify this by looking in the items XML, on Plex, and looking for this section:

```
<Guid id="imdb://tt0095016"/>
<Guid id="tmdb://562"/>
<Guid id="tvdb://324"/>
```
1077280 is the correct TMDB id that should be showing for the movie Die Hart. [TMDB Link](https://www.themoviedb.org/movie/1077280-die-hart)

We use this ID to tell Radarr/Sonarr/Overseerr/etc. exactly which movie or show we are talking about. If it is incorrect, then the request to those services fails as it does not exist. Technically it does, but with a different TMDB id than what Maintainerr got from Plex.

> `2024-12-26T20:14:45.572031605Z [maintainerr] | 26/12/2024 21:14:45  [DEBUG] [OverseerrGetterService] Couldn't find tmdb id for media 'Die Hart' with id '79387'. As a result, no Overseerr query could be made.`

TL;DR The best course of action is to try and use Fix Match on the item in Plex and see if it fixed the incorrect ID that was attached to the item.

**Comment by daerup** [comment]
Thanks to your comment I looked more into my metadata and as it turns out, I haven't refreshed my metadata for quite some time. Some movies were still in the old plex agent format, which doesn't work as expected with Maintainerr. After I refreshed my library everything seems to work, thanks!
